crates/wasi-provider/src/states/running.rs

```diff
@@ -0,0 +1,53 @@
+use k8s_openapi::api::core::v1::Pod as KubePod;
+use kube::api::{Api, PatchParams};
+use kubelet::container::Status;
+use kubelet::state::prelude::*;
+use log::error;
+
+use super::error::Error;
+use super::terminated::Terminated;
+use crate::PodState;
+
+state!(
+    /// The Kubelet is running the Pod.
+    Running,
+    PodState,
+    {
+        let client: Api<KubePod> = Api::namespaced(
+            kube::Client::new(pod_state.shared.kubeconfig.clone()),
+            pod.namespace(),
+        );
+        let mut completed = 0;
+        let total_pods = pod.containers().len();
+
+        while let Some((name, status)) = pod_state.run_context.status_recv.recv().await {
+            if let Err(e) = client
+                .patch_status(
+                    pod.name(),
+                    &PatchParams::default(),
+                    serde_json::to_vec(&status.to_kubernetes(name))?,
+                )
+                .await
+            {
+                error!("Unable to patch status, will retry on next update: {:?}", e);
+            }
+            if let Status::Terminated {
+                timestamp: _,
+                message,
+                failed,
+            } = status
+            {
+                if failed {
+                    return Ok(Transition::Error(Box::new(Error { message })));
+                } else {
+                    completed += 1;
+                    if completed == total_pods {
+                        return Ok(Transition::Advance(Box::new(Terminated)));
+                    }
+                }
+            }
+        }
+        Ok(Transition::Advance(Box::new(Terminated)))
```

Comment:
I think technically the states are Completed and CrashLoopBackoff.

Reply:
Good catch.